feat(databases-collections): Create a time-series collection from create database COMPASS-4860

Anemy opened the pull request:
COMPASS-4860 also gets COMPASS-3921

There's a large diff since we're moving some things around, so I'll leave some comments in places where things have changed.
Here are some of the changes:
- Moved `compass-collection-ddl` and `compass-database-ddl` into one package, currently it's named `databases-collections`, if y'all think of another name please let me know!
- Updated the `create database` and `create collection` modals to use [leafygreen](https://github.com/mongodb/leafygreen-ui) components. They do look a bit different now, still syncing with Claudia to finalize.
- Updated the electron environment for running the collections and databases views and modals in isolated development (see screenshot at the bottom).
- Updated how the create database modal was showing fields for collection properties so it now shares this with the create collection modal.
- Built on Maurizio's work on how the form for creating collections works, adding the `collection-fields` components. This involves removing a number of things from the store and moving them into this new component's state.

Some notes/thoughts for possible future things:
- We're adding in some leafygreen components here, which updates the look of components which exist in other forms in other places in Compass. This means we'll be losing some design consistency in the short term, until we have more things using these components. We should have a common place for these components (this has been `[hadron-react-components](https://github.com/mongodb-js/compass/tree/master/packages/hadron-react-components/src)` in the past). Probably something we want to discuss as a team on how we want to structure those, and style them (css in js with emotion and typescript? keep them how they are?`compass-components`?). 
- The `collections-table` and `databases-table` have a similar toolbar, however it's styles are duplicated. It would be nice to have this shared. Probably a `Toolbar` component could do us good.
- The `collation` component is used by compass-indexes. We aren't touching that part in this work, but it would be nice to reduce that code duplication (maybe when we make all of these modals look similar).
- Maurizio noticed in the [first pr for this work](https://github.com/mongodb-js/compass/pull/2196) is that an error is returned when creating collections with custom collation on only in the development environment (this works successfully in production/compass in video lower down):
<img width="400" alt="Screen Shot 2021-06-02 at 4 07 36 PM" src="https://user-images.githubusercontent.com/1791149/120545439-a71e9780-c3bc-11eb-9d26-d7c0d5559ea0.png">

**_Vids_**
___
*Create db with collection with collation and time series*

https://user-images.githubusercontent.com/1791149/120552104-c91c1800-c3c4-11eb-8326-718ca40c8b34.mp4


*Create collection with time series*

https://user-images.githubusercontent.com/1791149/120552089-c4576400-c3c4-11eb-8315-2499a2c26f01.mp4


*Screenshot of create db when connected to < 5*
<img width="651" alt="Screen Shot 2021-06-02 at 5 05 28 PM" src="https://user-images.githubusercontent.com/1791149/120552061-bdc8ec80-c3c4-11eb-8381-4619311fefe2.png">


~**Still working on a few tests for `collection-fields.jsx` and finalizing styles w/ Claudia, I figured this could be open while that's going on since I won't get to it till later in the day EU.**~ done

Since this is a new package, and we are removing old packages, we'll need to release on npm and update Compass' plugins when we're comfortable with these changes/ready to merge.

Anemy commented on packages/databases-collections/src/index.js:
```diff
@@ -0,0 +1,99 @@
+  appRegistry.registerRole('Global.Modal', DROP_DATABASE_ROLE);
+  appRegistry.registerStore('DatabasesPlugin.DatabasesStore', DatabasesStore);
+  appRegistry.registerStore('DatabasesPlugin.CreateDatabaseStore', CreateDatabaseStore);
+  appRegistry.registerStore('DatabasesPlugin.DropDatabaseStore', DropDatabaseStore);
```
Here's where we combined where the two plugins register to Compass.

Anemy commented on packages/databases-collections/LICENSE:
```diff
@@ -0,0 +1,201 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
```
Ah this might be outdated now, need to check

Anemy commented on packages/databases-collections/src/components/collection-fields/collection-fields.jsx:
```diff
@@ -0,0 +1,312 @@
+  }
+}
+
+export default class CollectionFields extends PureComponent {
```
Here's the `CollectionFields` component where we render the options for creating collections (shared w/ create database). 
The more general components above are things we probably want to have in a generalized place in Compass where we can also pull the theme.

Anemy commented on packages/databases-collections/src/components/create-database-modal/create-database-modal.jsx:
```diff
@@ -0,0 +1,160 @@
+        />
+        {this.renderCollectionNameRequiredNotice()}
+        {this.renderError()}
+      </ConfirmationModal>
```
Here's the updated `create-database-modal`. Here's what it used to be: https://github.com/mongodb-js/compass/blob/master/packages/compass-databases-ddl/src/components/create-database-modal/create-database-modal.jsx

gribnoysup commented on packages/databases-collections/src/components/create-database-modal/create-database-modal.jsx:
```diff
@@ -0,0 +1,163 @@
+          openLink={this.props.openLink}
+        />
+        {this.renderCollectionNameRequiredNotice()}
+        {this.renderError()}
```
This error probably needs some padding:

<img width="531" alt="image" src="https://user-images.githubusercontent.com/5036933/121497039-5241d900-c9db-11eb-809e-636775de43d6.png">

Anemy replied:
nice catch. Also how did you come across this error? Dataservice is null? 😅 

Here's how it looks now:
<img width="590" alt="Screen Shot 2021-06-10 at 11 53 43 AM" src="https://user-images.githubusercontent.com/1791149/121557368-b320df80-c9e2-11eb-9c07-74db41522ee1.png">

gribnoysup replied:
Haha, I just started the electron app before starting my local mongodb 😆 Thanks for fixing this!

gribnoysup commented on packages/databases-collections/src/components/collection-fields/collection-fields.jsx:
```diff
@@ -0,0 +1,313 @@
+  render() {
+    return (
+      <FieldSet>
+        <Checkbox
```
When I run it locally, this checkbox looks pretty weird:

<img width="188" alt="image" src="https://user-images.githubusercontent.com/5036933/121502920-e2365180-c9e0-11eb-8d34-4bbffd22bab1.png">

gribnoysup replied:
We figured it out! It's a zooming issue

gribnoysup commented on packages/databases-collections/src/components/collection-fields/collection-fields.jsx:
```diff
@@ -0,0 +1,313 @@
+            }`}
+            onChange={(val) => this.setCollationField(element.field, val)}
+            usePortal={false}
+            size={Size.Small}
```
A bit weird to me that we have TextInput of normal size, but Select of size small, they are noticeably different in size when both are visible:

<img width="226" alt="image" src="https://user-images.githubusercontent.com/5036933/121503066-03973d80-c9e1-11eb-912e-fd3fd13a7bfa.png">

Is that intentional?

Anemy replied:
This was intentional, the two thoughts behind it are the normal sized selects looked a bit large and there are so many collation options it they quickly took up a lot of space. Happy to put them back to normal size, did sync with Claudia on this and she was cool with both if I remember correctly.

gribnoysup replied:
Ah, okay, this makes sense, thanks for adding some context!

gribnoysup reviewed (approved):
Overall looks really neat! A few things caught my eye so I left some comments here and there

Anemy commented on packages/databases-collections/src/components/collection-fields/collection-fields.jsx:
```diff
@@ -0,0 +1,313 @@
+  renderDatabaseNameField() {
+    return (<FieldSet>
+      <TextInput
+        autoFocus
```
Nice catch.  I think this is a bit of a special case because these fields are in a modal. I think we do want the modal to auto focus the content of the modal so someone using a motor control or pure keyboard can interact with the modal content immediately instead of having to navigate somewhere. I think the modal component does this for us though and we shouldn't have these inputs trying to do that functionality. removed 👍

gribnoysup replied:
Yeah, leafygreen seems to correctly lock focus inside the modal 👍

Anemy commented on packages/databases-collections/README.md:
```diff
@@ -0,0 +1,14 @@
+
+## License
+
+Apache 2.0
```
Looks like we might have a few more places which are still referencing the old license in their READMEs. I'll open up a PR after this updating readmes of other packages (updated this package in this pr). Niice find.

Anemy commented on packages/databases-collections/LICENSE:
```diff
@@ -0,0 +1,201 @@
+Apache License
```
removed 👍

Anemy commented on packages/databases-collections/src/components/collection-fields/collection-fields.jsx:
```diff
@@ -0,0 +1,313 @@
+    };
+  }
+
+  renderDatabaseNameField() {
```
Good thinking. I haven't done much straight function usage with react in the past, I like the sound of it. Updated the PR to use some. Also broke out the two generic components, `collapsible-field-set` and `field-set`, which will make the next step of putting them in a more central place for Compass (let's chat about this next few days maybe - my current thinking is having a `compass-components` that's typescript which has generic components and light leafygreen wrappers. it would be plugged into a store or some global configurable for styles).

Anemy commented:
@gribnoysup pushed some updates if you'd like to take a look. 
- Moved some of the component parts into functions, see comment above for more.
- Updated the `collations` components, so now the new styles are shared with the `create-index-modal` in the `compass-indexes` package. (tested making a new index with collation and it seems to work - screenshot at bottom [1])
- Updated the license (removed it from the package and removed the old license part from the README)
Before merging I'll just make sure we're getting the most recent versions of package dependencies for these two repos we are merging from the main branch. 

[1]
<img width="558" alt="Screen Shot 2021-06-10 at 7 08 33 PM" src="https://user-images.githubusercontent.com/1791149/121608114-4b3cba00-ca1f-11eb-858b-1e062397cc69.png">

gribnoysup reviewed (approved):
Absolutely awesome work 🔥 Thanks for addressing those comments!

If you want any help with merging/releasing this thing, I'm happy to pair on it (we can test the whole thing locally before publishing to remote)
